### PR TITLE
Ati ros2 wrench publisher

### PR DESCRIPTION
**Pull request by EAkita**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description
* Added a class/node to publish wrenches as geometry and standard msgs 
* Added a config file with necessary params
* Updated the code format with clang-format
* Updated the README file to reflect new changes 

## How I Tested
* Currently, no script testing has been done. However, I successfully ran and visualized the published data on both topics, which is more than necessary for a class this small.  

[//]: # "Explain how you tested your changes"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+To launch the node, run: 
+  ```
+  ros2 run ati_ft_sensor ati_ft_sensor_test_sensor
+  ```
+#### Ati Wrench Publisher
+To publish wrenches on a /wrench_data topic, update the ip_address and the frame_id parameters in the config/ft_publisher.yaml file and run:
+ros2 launch ati_ft_sensor ati_ft_sensor.launch.py
+
+#### Faked Controller Wrench Publisher for Admittance Control
+To control a visualization/simulation of a ROS2-enabled robot with data from the ati sensor, set the pub_faked_wrench parameter to True and launch the node. This will publish an additional Float64MultiArray std msg for admittance control.
```
Comment (FabianEP11) [inline README.md]: You could include the name of the topic for clarity. (faked_forces_controller/commands ?)

File: config/ft_publisher.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ati_wrench_publisher:
+  ros__parameters:
+    frame_id: "end_effector_link"
+    pub_faked_wrench: True
+    ip_address: "192.168.1.84"
```
Comment (FabianEP11) [inline config/ft_publisher.yaml:5]: is that the default value?

File: nodes/ati_ft_sensor_wrench_pub.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+class ATIWrenchPublisher : public rclcpp::Node {
+  // Constructor
+ public:
+  ATIWrenchPublisher() : Node("ati_wrench_publisher") {
+    initializeNode();
+
+    // Create a timer that calls the publish function every millisecond.
+    timer_ = this->create_wall_timer(
+        std::chrono::milliseconds(1),
```
Comment (FabianEP11) [inline nodes/ati_ft_sensor_wrench_pub.cpp]: Call the function every millisecond?  Why do you need such high rate... 
That is like if you were expecting to publish at 1000 Hz

File: nodes/ati_ft_sensor_wrench_pub.cpp [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  std::string frame_id_, ip_address_;
+  bool fake_controller_wrench_;
+
+  void initializeNode() {
+    // Get the required parameters
+    this->declare_parameter("frame_id", rclcpp::PARAMETER_STRING);
+    this->declare_parameter("ip_address", rclcpp::PARAMETER_STRING);
+    this->declare_parameter("pub_faked_wrench", rclcpp::PARAMETER_BOOL);
+
+    rclcpp::Parameter frame_id = this->get_parameter("frame_id");
```
Comment (FabianEP11) [inline nodes/ati_ft_sensor_wrench_pub.cpp]: I think you can rewrite this on a more compact version.
frame_id_ = this->get_parameter("frame_id"); 
I dont think you need to load the param to get it again after, and then create a variable to store it as string.  (Same for the other params)

File: nodes/ati_ft_sensor_wrench_pub.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    ip_address_ = ip_address.as_string();
+    fake_controller_wrench_ = pub_faked_wrench.as_bool();
+
+    // Create a publisher for the sensor data topic.
+    pub_wrench_ = this->create_publisher<geometry_msgs::msg::WrenchStamped>(
+        "wrench_data", rclcpp::SensorDataQoS());
+
+    if (fake_controller_wrench_)
+      pub_fake_controller_wrench_ =
+          this->create_publisher<std_msgs::msg::Float64MultiArray>(
```
Comment (FabianEP11) [inline nodes/ati_ft_sensor_wrench_pub.cpp:71]: I don't fully understand why you need a different type of msg to publish the fake control wrench...  Why don't you use the same wrenchStamped or even just wrench

File: nodes/ati_ft_sensor_wrench_pub.cpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    wrench_msg.wrench.torque.x = torque[0];
+    wrench_msg.wrench.torque.y = torque[1];
+    wrench_msg.wrench.torque.z = torque[2];
+    pub_wrench_->publish(wrench_msg);
+
+    if (fake_controller_wrench_) {
+      auto msg = std::make_unique<std_msgs::msg::Float64MultiArray>();
+      msg->data = {force[0],  force[1],  force[2],
+                   torque[0], torque[1], torque[2]};
+      pub_fake_controller_wrench_->publish(std::move(msg));
```
Comment (FabianEP11) [inline nodes/ati_ft_sensor_wrench_pub.cpp:98]: Same comment here... 
I guess the robot is subscribing to this fake topic... Is that a default behavior or something you have to manually configure on the sim. Thats the reason I prefer wrench msg over the multiArray.

File: package.xml [hunk 7]
```diff
@@ -22,6 +22,8 @@
     <buildtool_depend>ament_cmake</buildtool_depend>
     <!-- CMake module containing helper to find the OS, boost, python, etc. -->
     <build_depend>mpi_cmake_modules</build_depend>
+    <build_depend>rclcpp</build_depend>
+    <build_depend>geometry_msgs</build_depend>
```
Comment (FabianEP11) [inline package.xml:26]: std::msgs  ?

**Review by FabianEP11 — COMMENTED** [review]
Looks fine to me. Minor comments

File: nodes/ati_ft_sensor_wrench_pub.cpp (hunk 5 above)
Comment (EAkita) [reply nodes/ati_ft_sensor_wrench_pub.cpp:71]: That's the msg type that the fake_controller accepts

**Pull request merged**